forms/tests/test_serializers.py

```diff
@@ -202,6 +212,58 @@ def test_company_id_validator(
     assert val_error.value.args[0]["non_field_errors"][0].code == "invalid_company_id"
 
 
+@pytest.mark.django_db
+def test_control_share(
+    django_db_setup, basic_template_form, plot_search_target, admin_user
+):
+    entries = {}
+    for section in basic_template_form.sections.filter(parent__isnull=True):
+        entries[section.identifier] = dict()
+        generate_entries(section, entries[section.identifier])
+    entries["hakijan-tiedot"]["sections"]["company-information"]["sections"][
+        "contact-person"
+    ]["fields"]["henkilotunnus"] = {"value": "010181-900C", "extraValue": ""}
+    entries["hakijan-tiedot"]["sections"]["company-information"]["fields"][
+        "y-tunnus"
+    ] = {"value": "1234567-8", "extraValue": ""}
+    entries["hakijan-tiedot"]["sections"]["company-information"]["fields"][
+        "hallintaosuus"
+    ] = {"value": "1/3", "extraValue": ""}
+    entries["hakijan-tiedot"]["sections"].update(
+        {
+            "company-information[0]": {
+                "fields": {
+                    "hallintaosuus": {"value": "2/3", "extraValue": "",},  # noqa: E231
+                }
+            }
+        }
+    )
+
+    answer_data = {
+        "form": basic_template_form.id,
+        "user": admin_user.id,
+        "entries": entries,
+        "targets": [plot_search_target.pk,],  # noqa: E231
+    }
+
+    answer_serializer = AnswerSerializer(data=answer_data)
+    # test that a correctly formatted ssn passes the validator
+    assert answer_serializer.is_valid()
+
+    answer_data["entries"]["hakijan-tiedot"]["sections"]["company-information[0]"][
+        "fields"
+    ]["hallintaosuus"] = {"value": "1/3", "extraValue": ""}
+    answer_serializer = AnswerSerializer(data=answer_data)
+
+    # test that a incorrectly formatted ssn is caught by the validator
```

Comment:
# test that control share validator returns validation error if the sum is not 1